lib/application.rb

```diff
@@ -1,12 +1,90 @@
-class LibraryApp
-  attr_accessor :db
+class Application
+  attr_accessor :current_user, :admin_screen
 
   def initialize
-    self.db = DatabaseConnector.new
+    self.current_user = nil
+    self.admin_screen = AdminScreen.new
   end
 
   def start
+    display_welcome_message
+    login_or_register_menu
+  end
+
+  private
+
+  def display_welcome_message
+    system('clear')
     puts "Welcome to the Library Management System!"
     puts "-----------------------------------------"
   end
+
+  def login_or_register_menu
+    choice = 0
+    while choice != 3
+      puts "Please select an option:"
+      puts "1. Login"
+      puts "2. Register"
+      puts "3. Exit"
+      print "Please enter your choice : "
+      choice = gets.chomp.to_i
+      display_welcome_message
+      case choice
+      when 1
+        login_user
+      when 2
+        register_user
+      when 3
+        puts "Goodbye! Exiting the Library Management System.\n"
+        exit(0)
+      else
+        puts "Invalid choice please enter again."
+      end
+    end
+  end
+
+  private
+
+  def login_user
+    puts "Login"
+    puts "-----------------------------------------"
+    print "Enter username : "
+    username = gets.chomp
+    print "Enter password : "
+    password = gets.chomp
+
+    user = UserController.find_by_username(username)
+    if user && BCrypt::Password.new(user.password) == password
```

Comment:
Think like this: If a check is related to user, it should be part of user class. Example: This can be written as
```ruby
if user && user.valid_password?(password)
  # do stuff
end

# inside user class
class User 
  # constructor and attrs
  def valid_password?(password_to_check)
    BCrypt::Password.new(password) == password_to_check
  end
end

```